packages/database_v2/artifacts/init-tables.sql

```diff
@@ -0,0 +1,316 @@
+    -- for this resource
+    -- {
+    --    status: 'ContentLinked'
+    --    uri:string
+    --    status_text:string
+    --    ipfs_uri: string|null
+    --    content_cid: string
+    --  }
+    'ContentLinked'
+);
+
+
+-- As per EIP-721 token asset is identified by a `tokenURI` that MAY
+-- point to a JSON file that conforms to the "ERC721 Metadata JSON Schema.
+-- Here we represent nft asset by a state machine in which states are identified
+-- by a `status` field from this enum.
+DROP TYPE IF EXISTS nft_asset_status CASCADE;
+CREATE TYPE nft_asset_status AS ENUM (
+    -- nft asset was created from `tokenURI` is queued for processing.
+    -- {
+    --    status: 'Queued'
+    --    token_uri: string
+    --    status_text: string
+    -- }
+    'Queued',
+    -- Failed to parse uri (it is invalid or maybe protocol isn't supproted)
+    -- {
+    --    status: 'URIParseFailed'
+    --    token_uri: string
+    --    status_text: string
+    --  }
+    'URIParseFailed',
+    -- Failed to fetch content from `token_uri`. It may have `ipfs_uri` if it
+    -- was possible to derive it from the `token_uri`.
+    -- {
+    --    status: 'ContentFetchFailed'
+    --    token_uri: string
+    --    status_text: string
+    --    ipfs_uri: string|null
+    -- }
+    'ContentFetchFailed',
+    -- Failed to parse content from `token_uri` as per ERC721 Metadata JSON
+    -- schema.
+    -- {
+    --    status: 'ContentParseFailed'
+    --    token_uri: string
+    --    status_text: string
+    --    ipfs_uri: string|null
+    -- }
+    'ContentParseFailed',
+    -- Resource that we failed to submit pin request for. If it has `ipfsURI`
+    -- we've send a request to a cluster asking to pin given cid and it may have
+    -- failed. If we could not derive ipfsURI we've tried fetching data and then
+    -- we've send it to ipfs cluster to add to the network and pin, but cluster
+    -- failed. It also maybe that cluster took too long to respond so we've
+    -- aborted request. In any case this indicates some issue with cluster.
+    -- {
+    --    status: 'PinRequestFailed'
+    --    token_uri: string, statusText: string
+    --    status_text: string
+    --    ipfs_uri: string|null
+    --  }
+    'PinRequestFailed',
+    -- Asset was fetched, parsed and linked to a corresponding metadata record.
+    -- It has `content_cid`  that cluster was asked to pin.
+    -- {
+    --    status: 'Linked'
+    --    uri:string
+    --    status_text:string
+    --    ipfs_uri: string|null
+    --    content_cid: string
+    --  }
+    'Linked'
+);
+
+DROP TYPE IF EXISTS pin_status CASCADE;
+CREATE TYPE pin_status AS ENUM (
+    'PinError',
+    'PinQueued',
+    'Pinned',
+    'Pinning'
+);
+
+DROP TYPE IF EXISTS pin_service CASCADE;
+CREATE TYPE pin_service AS ENUM (
+    'Pinata',
+    'IpfsCluster'
+);
+
+-- A blochain block identified by a it's hash.
+CREATE TABLE IF NOT EXISTS blockchain_block (
+    hash TEXT NOT NULL,
+    number INTEGER NOT NULL,
```

Comment:
I imagine this needs to be `BIGINT` too?

Reply:
I'm not sure where exactly this type is defined by the thegraph.com but this table corresponds to following type

<img width="234" alt="Screenshot 2021-09-17 at 1 55 46 PM" src="https://user-images.githubusercontent.com/21236/133852677-bbb768d0-70d1-433c-bff1-10ba95b4f492.png">

So I assumed it would be correspond to `INTEGER` here

Reply:
And according to [subgraph docs](https://thegraph.com/docs/developer/create-subgraph-hosted#graphql-supported-scalars) (quoting here)

> | Type | Description |
> |-- | --|
> |Bytes | Byte array, represented as a hexadecimal string. Commonly used for Ethereum hashes and addresses.|
> |ID | Stored as a string.|
> |String | Scalar for string values. Null characters are not supported and are automatically removed.|
> |Boolean | Scalar for boolean values.|
> |Int | The GraphQL spec defines Int to have size of 32 bytes.|
> |BitInt | Large integers. Used for Ethereum's uint32, int64, uint64, ..., uint256 types. Note: Everything below uint32, such as int32, uint24 or int8 is represented as i32.|
> |BigDecimal | BigDecimal  High precision decimals represented as a signficand and an exponent.  The exponent range is from −6143 to +6144. Rounded to 34 significant  digits.|

`Int` is 32 bytes which is wheres according to postgres docs `INTEGER` is only 4 bytes and `BIGINT` is 8 bytes. So it seems that even `BIGINT` may not be adequate here.

I think we can roll with `BIGINT` and migrate to another scalar if we do indeed reach the limit.

Reply:
Agree on the BIGINT, just changed everything to reflect this